src/System.Management.Automation/engine/CommandCompletion/PseudoParameterBinder.cs

```diff
@@ -1862,13 +1862,17 @@ private static bool IsTypeEquivalent(Type argType, Type paramType)
             while (unboundArgumentsIndex < unboundArgumentsCollection.Count)
             {
                 AstParameterArgumentPair argument = unboundArgumentsCollection[unboundArgumentsIndex++];
-                if (!argument.ParameterSpecified)
+                bool splatted = (argument is AstPair a) && (a.Argument is VariableExpressionAst v) && v.Splatted;
+                if (!argument.ParameterSpecified && splatted == false)
```

Comment:
Testing a boolean value for equality is unneeded; `if (x == false) { ... }` is the same as `if (!x) { ... }`.

```suggestion
                bool splatted = argument is AstPair astPair
                    && astPair.Argument is VariableExpressionAst argumentVariable
                    && argumentVariable.Splatted;
                if (!splatted && !argument.ParameterSpecified)
```